nano/node/election.cpp

```diff
@@ -693,24 +695,9 @@ std::vector<nano::vote_with_weight_info> nano::election::votes_with_weight () co
 
 nano::stat::detail nano::to_stat_detail (nano::election_behavior behavior)
 {
-	switch (behavior)
-	{
-		case nano::election_behavior::normal:
-		{
-			return nano::stat::detail::normal;
-		}
-		case nano::election_behavior::hinted:
-		{
-			return nano::stat::detail::hinted;
-		}
-		case nano::election_behavior::optimistic:
-		{
-			return nano::stat::detail::optimistic;
-		}
-	}
-
-	debug_assert (false, "unknown election behavior");
-	return {};
+	auto value = magic_enum::enum_cast<nano::stat::detail> (magic_enum::enum_name (behavior));
```

Comment:
I am not convinced this is an improvement overall, although the code brevity is a nice thing.
If a new behaviour is added that doesn't exist as a detail then the problem will be silently ignored whereas, in the previous case, the compiler would complain that not all values were handled by the switch case statement.
I am not objecting to the change, just wondering what this gives us, other than a few less lines of very obvious code.

Reply:
I hope you're joking... This used to be and still is one on the most annoying and unnecessary things to do in this codebase.